- added a secondary  color for highlight and annotation  text

cage2 opened the pull request:
The two colors are  shown in an alternate  way for annotation placed  on the
  same lines;
- fixed point position when text was added on a line with annotation box
``` text
                                       +--- you add text here...
                                       |
                                       V
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
                                     +---------+
^ but the cursor is shown here       | note    |
| instead :-(                        | note    |
|                                    | note    |
|                                    +---------+
|
|
```
- fixed annotation position when a single line (as region) is annotated.

Commit:
- added a secondary  color for highlight and annotation  text, the two
  colors are  shown in an alternate  way for annotation placed  on the
  same lines;
- fixed point position when text was added on a line with annotation box

                                       +--- you add text here...
                                       |
                                       V
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
                                     +---------+
^ but the cursor is shown here       | note    |
| instead :-                         | note    |
|                                    | note    |
|                                    +---------+
|
|

- fixed annotation position when a single line (as region) is annotated.

bastibe commented:
I don't quite understand. Could you post a screen shot of what this looks like?

cage2 commented:
Hi!

Sorry for the inconvenience, :( here are two links: the first for the colors feature and a little GIF for the bug. For the latter notes that the cursor does not disappears (this is an artifact of  the software i used to record the video) but is still displayed under the the 'T' letter of 'this'.

https://www.autistici.org/interzona/img/misc/colors.png
https://www.autistici.org/interzona/img/misc/bug.gif

Bye!
C.

bastibe commented on annotate.el:
```diff
@@ -173,11 +183,38 @@ major mode is a member of this list (space separated entries)."
          ,@body)
      (setf inhibit-modification-hooks t)))
 
+(defun annotate-calc-end-of-line ()
```
I would call this `annotate-end-of-line-pos`, similar to the original `end-of-line`, but with an added `pos` to indicate that the cursor is not moved.

cage2 replied:
sure!

bastibe commented on annotate.el:
```diff
@@ -173,11 +183,38 @@ major mode is a member of this list (space separated entries)."
+    (end-of-line)
+    (point)))
+
+(defun annotate-calc-beginning-of-line ()
```
I would call this `annotate-beginning-of-line-pos`, similar to the original `beginning-of-line`, but with an added `pos` to indicate that the cursor is not moved.

cage2 replied:
ok!

bastibe commented on annotate.el:
```diff
@@ -665,30 +703,40 @@ to 'maximum-width'."
+          (eol             (progn (end-of-line) (point)))
+          (text            "")
+          (overlays        nil)
+          (face-type-count 1))
```
why is it a face-*type*-count? I think I would simply call this `annotation-counter` or even just `idx`.

cage2 replied:
``idx`` look too general to me, ``annotation-counter`` looks fine, though! :)

bastibe commented on annotate.el:
```diff
@@ -173,11 +183,38 @@ major mode is a member of this list (space separated entries)."
+    (beginning-of-line)
+    (point)))
+
+(defun annotate-before-change-fn (a b)
```
Missing docstring.

cage2 replied:
Right! fixed.

bastibe commented on annotate.el:
```diff
@@ -82,11 +82,21 @@
   :group 'annotate)
 
+(defface annotate-annotation-secondary
+  '((t (:background "turquoise" :foreground "black")))
```
This is a very stark contrast. I would go with something more in the orange-yellow spectrum, akin to the alternating line highlights in tables.

cage2 replied:
I chose the (more or less) complementary color, but without a valid reason actually! :)
I changed it to khaki, look nice to me anyway if you do not like it just tell me which one you prefer and i will add it. :)

bastibe reviewed (changes requested):
OK, cool, I understand now. Thank you for providing the screenshots!

I like this very much!

I've left some comments for discussion. Most of these are pretty minor. The one most important to me is the color, honestly.

Commit:
- changed functions name
  - annotate-calc-end-of-line       -> annotate-end-of-line-pos;
  - annotate-calc-beginning-of-line -> annotate-beginning-of-line-pos.

Commit:
- changed variable name
  - face-type-count -> annotation-counter

Commit:
- added docstrings.

Commit:
- changed secondary color.

Commit:
- fixed typos.

bastibe commented:
OK, cool! Looks good to me!

Are you ready to merge?

Commit:
- fixed bug with multiline annotation

  when the  user annotated some  text placed  before (but on  the same
  line) an  already annotated  multiline region the latter annotation
  was displayed twice, This patch mitigate this issue.

cage2 commented:
On Mon, Oct 21, 2019 at 02:21:21AM -0700, Bastian Bechtold wrote:

Hello bastibe!

> OK, cool! Looks good to me!

Oh thank you! Thanks  to this package and with your help  I feel a bit
more confident with elisp now. :)

> Are you ready to merge?

With this latest patch just pushed i think the PR could be merged.

Bye!
C.

bastibe commented:
Well then, thank you very much for another great contribution!

Say, would you like to become co-maintainer of this project? It wouldn't come with any obligations at all. But you seem interested and able to keep this project alive.

cage2 commented:
> Well then, thank you very much for another great contribution!

My pleasure!
 
> Say, would you like to become co-maintainer of this project? It wouldn't come with any obligations at all. > But you seem interested and able to keep this project alive.

Sure! I found this project useful and i am using it on a regular basis (for annotate source code to be honest, i would like to make it works with org-mode but seems i have to study a lot more for that :-( ).

Anyway i think i could help maintaining, so i accept with joy your proposal! :)
Bye!
C.

bastibe commented:
Great! I've just given you commit rights.

It's still a good idea to discuss big changes in pull requests, but you may now commit bug fixes etc. directly to the master if you like!

cage2 commented:
On Tue, Oct 22, 2019 at 11:33:06PM -0700, Bastian Bechtold wrote:

Hello!

> Great! I've just given you commit rights.

Thank you very much!


> It's still a good idea to discuss big changes in pull requests,

Sure i also prefer this way of contribute to the repository!

> but you may now commit bug fixes etc. directly to the master if you like!

Very well! I hope to help improving this package in the future! :)

Bye!
C.
